packages/asgardex-cosmos/src/util.ts

```diff
@@ -0,0 +1,34 @@
+import { MsgMultiSend, MsgSend } from 'cosmos-client/x/bank'
+import { Msg } from 'cosmos-client'
+import { Asset, assetFromString } from '@thorchain/asgardex-util'
+import { AssetAtom, AssetMuon, CosmosChain } from './cosmos/types'
+
+/**
+ * Type guard for MsgSend
+ */
+export const isMsgSend = (v: Msg): v is MsgSend => 
+  (v as MsgSend)?.amount !== undefined && (v as MsgSend)?.from_address !== undefined && (v as MsgSend)?.to_address !== undefined
+
+/**
+ * Type guard for MsgMultiSend
+ */
+export const isMsgMultiSend = (v: Msg): v is MsgMultiSend => 
+  (v as MsgMultiSend)?.inputs !== undefined && (v as MsgMultiSend)?.outputs !== undefined
+
+/**
+ * Get denom from Asset
+ */
+export const getDenom = (v: Asset): string => {
+  if (v === AssetAtom) return 'uatom'
+  if (v === AssetMuon) return 'umuon'
```

Comment:
It's safer to compare strings of assets here ^:

```
assetToString(v) === assetToString(AssetAtom)
// or
assetToString(v) === assetToString(AssetMuon)
```
In other cases `===` will force you to use the same object and it will fail if not ...